index.js

```diff
@@ -216,6 +215,7 @@ NodeListIterator.prototype.next = function () {
  * @return {Node} The heap's new minimum node.
  */
 function cut(node, parent, minNode, compare) {
+  node.parent = undefined;
```

Comment:
I'd love a test case which fails before a passed after for this, I can write it if you can provide some reliable steps to get `decreaseKey` to fail.